cmd/podman/umount.go

```diff
@@ -24,18 +37,60 @@ func umountCmd(c *cli.Context) error {
 	}
 	defer runtime.Shutdown(false)
 
+	umountAll := c.Bool("all")
 	args := c.Args()
-	if len(args) == 0 {
+	if len(args) == 0 && !umountAll {
 		return errors.Errorf("container ID must be specified")
 	}
-	if len(args) > 1 {
-		return errors.Errorf("too many arguments specified")
+	if len(args) > 0 && umountAll {
+		return errors.Errorf("when using the --all switch, you may not pass any container IDs")
 	}
 
-	ctr, err := runtime.LookupContainer(args[0])
-	if err != nil {
-		return errors.Wrapf(err, "error looking up container %q", args[0])
-	}
+	umountContainerErrStr := "error unmounting container"
+	var lastError error
+	if len(args) > 0 {
+		for _, name := range args {
+			ctr, err := runtime.LookupContainer(name)
+			if err != nil {
+				if lastError != nil {
+					logrus.Error(lastError)
+				}
+				lastError = utils.WriteError(os.Stderr, errors.Wrapf(err, "%s %s", umountContainerErrStr, name), lastError)
+				continue
+			}
+
+			if err = ctr.Unmount(); err != nil {
+				if lastError != nil {
+					logrus.Error(lastError)
+				}
+				lastError = utils.WriteError(os.Stderr, errors.Wrapf(err, "%s %q", umountContainerErrStr, name), lastError)
+				continue
+			}
+			fmt.Printf("%s\n", ctr.ID())
+		}
+	} else {
+		containers, err := runtime.GetContainers()
+		if err != nil {
+			return errors.Wrapf(err, "error reading Containers")
+		}
+		for _, ctr := range containers {
+			mountpoint, err := ctr.Mountpoint()
```

Comment:
I don't know if I would consider this a safe way of identifying mounted containers. If you need to do this, add a way to access container.state.IsMounted